doc/user/content/transform-data/patterns/percentiles.md

```diff
@@ -10,134 +10,306 @@ menu:
+The following example assumes you have not previously created and populated the
+`input` table from the [Using histograms to compute percentiles
+example](#example). If you have created and populated the table, skip to step 3.
 
-HDR histograms are related to how [floating point numbers are represented](https://en.wikipedia.org/wiki/Double-precision_floating-point_format) as integers. The underlying assumption is that smaller numbers require a higher precision to be distinguishable (e.g. 5 ms and 6 ms are different and should be in different buckets) whereas larger numbers can be rounded more aggressively as their relative error becomes less relevant (e.g. 10000 ms and 10001 ms are almost the same and can reside in the same bucket).
+{{</ tip >}}
 
-The following snippets reduce the precision of numbers to limit the amount of required buckets. Numbers are decomposed into their floating point representation
+1. Create a table `input`:
 
-	n = sign * mantissa * 2**exponent
+   ```mzsql
+   CREATE TABLE input (value BIGINT);
+   ```
 
-and the precision of the mantissa is then reduced to compute the respective bucket.
+2. Insert into the `input` table values `1` to `10`.
 
-The basic ideas of using histograms to compute percentiles remain the same; but determining the bucket becomes more involved because it’s now composed of the triple (sign, mantissa, exponent).
+   ```mzsql
+   INSERT INTO input
+   SELECT n FROM generate_series(1,10) AS n;
+   ```
 
-```mzsql
--- precision for the representation of the mantissa in bits
-\set precision 4
+1. Create a `hdr_histogram` view. To reduce the number of buckets, the values
+   are first decomposed into their floating point representation (i.e.,
+   `mantissa * 2^exponent`), and with the precision of the mantissa is reduced
+   to 4, reconstructed.
 
-CREATE VIEW hdr_histogram AS
-SELECT
-  CASE WHEN value<0 THEN -1 ELSE 1 END AS sign,
-  trunc(log(2.0, abs(value)))::int AS exponent,
-  trunc(pow(2.0, :precision) * (value / pow(2.0, trunc(log(2.0, abs(value)))::int) - 1.0))::int AS mantissa,
-  count(*) AS frequency
-FROM input
-GROUP BY sign, exponent, mantissa;
-```
+   {{< tabs >}}
 
-The `hdr_distribution` view below reconstructs the `bucket` (with reduced precision), and determines the cumulative count and cumulative distribution.
+   {{< tab "Materialize Console" >}}
 
 ```mzsql
-CREATE VIEW hdr_distribution AS
+CREATE VIEW hdr_histogram AS
+WITH
+  input_parts AS (
+    SELECT
+      CASE WHEN value = 0 THEN NULL
+          ELSE trunc(log(2, abs(value)))::int
+      END AS exponent,
+      CASE WHEN value = 0 THEN NULL
+          ELSE value / pow(2.0, trunc(log(2, abs(value)))::int)
+      END AS mantissa
+    FROM input
+  ),
+  buckets AS (
+    SELECT
+      trunc(mantissa * pow(2.0, 4)) / pow(2.0, 4) -- precision is reduced to 4
```

Comment:
This is where I needed to go back and understand `mantissa`. I think this is saying something like
```
-- map (0.5, 1] to (8, 16], and round toward zero
```
which .. typing it out seems a bit weird because `16` only shows up if we had exactly a mantissa of `1.0` if I understand correctly. Not terrible, just "wasting" a bucket. Can ponder if we should fix, but perhaps it's not the most important thing.

Reply:
Hm, the way I think about this (I didn't write the logic, just cleaned it up) is that multiplying with `2^x` is a left shift by `x` and dividing is a right shift. So here we shift left by 4 bits, truncate to an integer, which clears all the bits beyond our selected precision of 4, and then shift back.

I'm not sure where we'd end up wasting a bucket in this. But I also don't understand where the `(0.5, 1]` interval comes from, as noted above I _think_ it should be `[1, 2)`.